client.go

```diff
@@ -127,6 +132,41 @@ func (c *Client) recvVersion() error {
 	return nil
 }
 
+func (c *Client) recv() {
+	sendAll := func(res result) {
```

Comment:
please make this a method, and give it a comment. It probably shouldn't be called `sendAll`, maybe `demux` or `dispatch` would be better